lib/astarte_vmq_plugin/config.ex

```diff
@@ -67,6 +68,38 @@ defmodule Astarte.VMQ.Plugin.Config do
     Application.put_env(:astarte_rpc, :amqp_connection, astarte_rpc_amqp_opts)
   end
 
+  defp init_ssl_options(amqp_options) do
+    ssl_options = Application.get_env(:astarte_vmq_plugin, :amqp_ssl, [])
+    build_ssl_options(ssl_options, amqp_options)
+  end
+
+  defp build_ssl_options([], amqp_options), do: amqp_options
+
+  defp build_ssl_options(ssl_options, amqp_options) do
+    case Keyword.fetch(ssl_options, :disable_sni) do
+      {:ok, false} ->
+        populate_ssl_options(ssl_options, amqp_options)
+
+      _ ->
+        amqp_options
+    end
+  end
+
+  defp populate_ssl_options(ssl_options, amqp_options) do
+    server_name_indication =
+      case Keyword.fetch(ssl_options, :custom_sni) do
+        {:ok, custom_sni} ->
+          custom_sni
+
+        _ ->
+          Keyword.fetch!(amqp_options, :host)
+      end
+
+    amqp_options
+    |> Keyword.put(:server_name_indication, server_name_indication)
+    |> Keyword.put(:verify, :verify_peer)
```

Comment:
These all go under `ssl_options` while you're putting them directly in amqp options